Lib/distutils/cygwinccompiler.py

```diff
@@ -112,36 +111,15 @@ def __init__(self, verbose=0, dry_run=0, force=0):
                 "Compiling may fail because of undefined preprocessor macros."
                 % details)
 
+        # For backwards compatibility, numpy uses this, but luckily has a fallback if None:
+        # https://github.com/numpy/numpy/blob/ff84c67ca41d/numpy/distutils/mingw32ccompiler.py#L67
+        self.gcc_version = None
+
         self.cc = os.environ.get('CC', 'gcc')
```

Comment:
not related to this PR, but should it default to `cc` instead of `gcc`?
```suggestion
        self.cc = os.environ.get('CC', 'cc')
```

Reply:
cc.exe seems like a MSYS2 only thing: https://github.com/msys2/MINGW-packages/blob/68b22c4a024bf434635441c6ce5ffc3e7cf8a220/mingw-w64-gcc/PKGBUILD#L319 so I'd rather not.

Reply:
oh ok.